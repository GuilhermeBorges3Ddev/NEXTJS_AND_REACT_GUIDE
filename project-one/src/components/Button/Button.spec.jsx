import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./index";

describe("<Button />", () => {
  it("should render the button with the text 'Load more'", () => {
    render(<Button text="Load More Posts..." />);
    expect.assertions(1);
    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });
  it("should call function on button click", () => {
    const fn = jest.fn();
    render(<Button text="Load More Posts..." loadMorePostsAction={fn} />);
    const button = screen.getByRole("button", { name: /load more posts/i });
    fireEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });
  it("should be disabled when disabled prop is true", () => {
    render(<Button text="Load More Posts..." isDisabled={true} />);
    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).toBeDisabled();
  });
  it("should be enabled when disabled prop is false", () => {
    render(<Button text="Load More Posts..." isDisabled={false} />);
    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).not.toBeDisabled();
  });
});
