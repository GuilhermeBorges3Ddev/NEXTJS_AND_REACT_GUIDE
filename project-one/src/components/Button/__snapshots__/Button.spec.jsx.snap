// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the Button tested with the snaphot 1`] = `
<button
  class="button"
>
  Load More Posts...
</button>
`;
